vassal-app/src/main/java/VASSAL/tools/io/ZipArchive.java

```diff
@@ -425,18 +427,50 @@ private void moveFile(Path src, Path dst) throws IOException {
         // attempt to copy to the destination
         Files.copy(src, dst, StandardCopyOption.REPLACE_EXISTING);
       }
-      catch (IOException e) {
-        // copy failed
-        final String fmt = "Unable to overwrite %s, so data written to %s instead: %s";
-        throw new IOException(
-          String.format(
-            fmt,
-            dst.toAbsolutePath(),
-            src.toAbsolutePath(),
-            e.getMessage()
-          ),
-          e
-        );
+      catch (IOException copyException) {
+        Path dst2 = dst.resolveSibling(src.getFileName()); // Our "temp" name but in the destination directory
+        final String fmt = Resources.getString("Editor.ZipArchive.overwrite") + "\n%s"; //NON-NLS
+        try {
+          // Try to at least put it in same directory as the module
+          Files.copy(src, dst2, StandardCopyOption.REPLACE_EXISTING);
```

Comment:
Should is be `Files.move` instead?

Reply:
I lean against it, because in this case we already tried a Files.move and it failed, and then we tried the Files.copy and that failed too. Obviously it is *possible* that a Files.move would now work but is not guaranteed. Given that we're already in the "more rare, error-handling case", adding *another* layer of try/catch in here seemed to needlessly complicate the flow & readability for a relatively minor performance saving in a relatively rare situation.

Reply:
One reason a second `Files.move` may succeed when the first failed is that the destination file shouldn't already exist for the second try, but this is fine.